flink-runtime/src/main/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElection.java

```diff
@@ -56,4 +56,40 @@ public boolean hasLeadership(UUID leaderSessionId) {
     public void close() throws Exception {
         parentService.remove(contenderID);
     }
+
+    /**
+     * {@link ParentService} is introduced to define the protocol between {@link
+     * LeaderElectionService} and {@link LeaderElection}.
+     */
+    abstract static class ParentService {
```

Comment:
```suggestion
    static abstract class ParentService {
```
the order felt strange. But it looks like we're not enforcing this special case with checkstyle. :thinking: Anyway, I checked other occurrences in the code. There's only one occurrence with `HiveMetaStoreClient.MetastoreMapIterable` where it is used. I propose to stick to the order where `static` comes first for consistency reasons

Reply:
Done.

Reply:
Looks like a was bad in grepping the code for this code snippet. There are plenty of occurrences for your way and there is a spotless rule for that order. I changed it back to make the CI build succeed.